### ISSUE TITLE
Due date shown 1 day earlier again

### ISSUE THREAD

**Issue by tolnai** [RogerSelwyn/MS365-ToDo#38]
I'm having trouble again with due dates not showing correctly.

Here is my selective bin collection entry in devtools, showing 11.20 (which is good)
![image](https://github.com/user-attachments/assets/70829ad6-4f48-4404-8486-f6ad892e29ac)

But then in the todo list it shows 11.19 (which is wrong)
![image](https://github.com/user-attachments/assets/9f8007c3-d293-4332-a310-d86e638e3c2b)

In the meanwhile we went from DST to non-DST, so perhaps something about time zones may impact this?

**Comment by RogerSelwyn** [comment]
Did you change your timezone to match on ha

**Comment by tolnai** [comment]
I didn't change anything, it is set based on geography.

![Screenshot_20241120_002518_Home Assistant](https://github.com/user-attachments/assets/3e7ed42d-f3ef-48cd-825d-ae3adc24fa80)

**Comment by RogerSelwyn** [comment]
Can you try adjusting the code at line 243 in classes/todo_integration as follows:


```
            if item.due:
                if item.due.time() != time(0, 0, 0):
                    due = item.due.astimezone(self._ha_timezone)
                    if due.time() != time(0, 0, 0) and due.time() > time(12, 0, 0):
                        due = due.date() + timedelta(days=1)
                    else:
                        due = due.date()
                else:
                    due = item.due.date()

                todo[ATTR_DUE] = due
```

It's difficult for me to test because I'm now at GMT+0.

**Comment by RogerSelwyn** [comment]
My error, I'd missed the point (in both sets of changes) of where it provides tasks for the in built HA core todo panel, as opposed to as an attribute on the entity.

[todo_integration.py.zip](https://github.com/user-attachments/files/17915239/todo_integration.py.zip)

Please replace todo_integration.py with the attached version.

**Comment by tolnai** [comment]
With the changes, it looks like dates appear fine. At least at the moment. :) Any other cases I should try/test?

**Comment by RogerSelwyn** [comment]
I don’t think so. I’m in the process of adding formal testing to the integration which I hope will cover it.

**Comment by RogerSelwyn** [comment]
Issue fixed with v1.0.0
